### PR TITLE
dealing with missing row in shoebox db

### PR DESCRIPTION
**Pull request by eishay**
Failed to ingest Shoebox Article updates from Rover. java.lang.Exception: failed processing normalization info for uri 6964029, of kind DefaultArticle NormalizationInfo(None,None,Set(),None) : http://tasker.dinglisch.net/guides.html
  Cause: java.util.NoSuchElementException: Invoker.first
  Cause: [No Cause]

Total Errors since start: 387
Total Errors since last alert: 30
Server: shoebox-spot-1
Server started at: 2015-05-24T21:03:27.861Z
Full Details

2015-05-25T03:32:33.926Z: [3a5ff8a3-8dc6-42f8-935b-8c385e5a2fa3]
error message [Failed to ingest Shoebox Article updates from Rover.]
Exception stack trace:
Cause:
java.lang.Exception: failed processing normalization info for uri 6964029, of kind DefaultArticle NormalizationInfo(None,None,Set(),None) : http://tasker.dinglisch.net/guides.html
com.keepit.shoebox.rover.NormalizationInfoIngestionHelper.liftedTree1$1(NormalizationInfoIngestionHelper.scala:35)
com.keepit.shoebox.rover.NormalizationInfoIngestionHelper.processNormalizationInfo(NormalizationInfoIngestionHelper.scala:27)
com.keepit.shoebox.rover.ShoeboxArticleIngestionActor.com$keepit$shoebox$rover$ShoeboxArticleIngestionActor$$processNormalizationInfo(ShoeboxArticleIngestionActor.scala:128)
com.keepit.shoebox.rover.ShoeboxArticleIngestionActor[a].apply(ShoeboxArticleIngestionActor.scala:117)
com.keepit.shoebox.rover.ShoeboxArticleIngestionActor[a].apply(ShoeboxArticleIngestionActor.scala:116)
scala.concurrent.Future[a].apply(Future.scala:251)
scala.concurrent.Future[a].apply(Future.scala:249)
scala.concurrent.impl.CallbackRunnable.run(Promise.scala:32)
akka.dispatch.BatchingExecutor$Batch[a].processBatch$1(BatchingExecutor.scala:67)
akka.dispatch.BatchingExecutor$Batch[a].apply$mcV$sp(BatchingExecutor.scala:82)
akka.dispatch.BatchingExecutor$Batch[a].apply(BatchingExecutor.scala:59)
akka.dispatch.BatchingExecutor$Batch[a].apply(BatchingExecutor.scala:59)
scala.concurrent.BlockContext$.withBlockContext(BlockContext.scala:72)
akka.dispatch.BatchingExecutor$Batch.run(BatchingExecutor.scala:58)
akka.dispatch.TaskInvocation.run(AbstractDispatcher.scala:41)
akka.dispatch.ForkJoinExecutorConfigurator$AkkaForkJoinTask.exec(AbstractDispatcher.scala:393)
scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)

### CODE REVIEW

**Comment by eishay** [comment]
Seems like its always the same uri #6964029.
This id does not exist in the db. One id before and one after does exist.

mysql> select \* from normalized_uri where id = 6964029\G
Empty set (0.00 sec)

@leogrim @stkem @andrewconner

File: kifi-backend/modules/shoebox/app/com/keepit/shoebox/rover/ShoeboxArticleIngestionActor.scala [hunk 1]
```diff
@@ -65,6 +65,18 @@ class ShoeboxArticleIngestionActor @Inject() (
...
+              Try(uriRepo.get(article.uriId)) match {
+                case Success(_) =>
+                  true
+                case Failure(e) =>
+                  airbrake.notify(s"article uri is not in the db: $article", e)
+                  false
+              }
+            }
+          }
           processRedirectsAndNormalizationInfo(updates).map { partiallyProcessedUpdatesByUri =>
```
Comment (andrewconner) [inline kifi-backend/modules/shoebox/app/com/keepit/shoebox/rover/ShoeboxArticleIngestionActor.scala:80]: Should this use `verifiedAtricles`?

Comment (eishay) [inline kifi-backend/modules/shoebox/app/com/keepit/shoebox/rover/ShoeboxArticleIngestionActor.scala:80]: yes! :-)

Comment (eishay) [inline kifi-backend/modules/shoebox/app/com/keepit/shoebox/rover/ShoeboxArticleIngestionActor.scala:80]: fix is on the way

**Comment by leogrim** [comment]
:+1: thanks, I deactivated a couple of entries from the `ArticleInfo` table last week to unstuck the plugin, I should have done something more systematic like this. Still no clue about what's going with the DB?

**Pull request merged**